File: python/src/doublePoint.py
from typing import List
# fast가 slow아 다르면 (중복구간을 벗어나면)slow 를 진행시킨다
def remove_duplicates(arr:List[int])->int:
    slow =0
    for fast in range(len(arr)):
        if arr[fast]!= arr[slow]:
            slow+=1
            arr[slow]=arr[fast]
    return slow +1

File: typescript/src/doublePoint.ts
import { Node } from "../basic/node.ts";

function removeDuplication<T>(arr: T[]) {
  let slow = 0;
  for (let fast = 0; fast < arr.length; fast++) {
    if (arr[fast] != arr[slow]) {
      slow++;
      arr[slow] = arr[fast];
    }
  }
}

function middleOfList<T>(head: Node<T>) {
  let slow: Node<T> | null = head;
  let fast: Node<T> | null = head;
  while (fast && fast.next) {
    fast = fast!.next.next;
    slow = slow!.next;
  }
  return slow?.val;
}
